File: Practice/src/coding_basic_training/CreateArray4.java
package coding_basic_training;

import java.util.Stack;

public class CreateArray4 {
//	배열 만들기4
	public int[] solution(int[] arr) {
		Stack<Integer> s = new Stack<Integer>();
		for (int i = 0; i < arr.length;) {
			if (s.size() == 0 || s.peek() < arr[i]) {
				s.add(arr[i++]);
			} else {
				s.pop();
			}
		}
		return s.stream().mapToInt(i -> i).toArray();
	}
}

File: Practice/src/coding_basic_training/CreateArray4.js
function solution(arr) {
    var stk = [];
    for (let i = 0; i < arr.length; ) {
        if (stk.length == 0 || stk[stk.length - 1] < arr[i]) {
            stk.push(arr[i++]);
        } else {
            stk.pop();
        }
    }
    return stk;
}
